etna/pipeline/base.py

```diff
@@ -270,10 +270,22 @@ def _forecast_prediction_interval(
             raise ValueError("Pipeline is not fitted! Fit the Pipeline before calling forecast method.")
         with tslogger.disable():
             _, forecasts, _ = self.backtest(ts=self.ts, metrics=[MAE()], n_folds=n_folds)
-        forecasts = TSDataset(df=forecasts, freq=self.ts.freq)
+
+        self._add_forecast_borders(backtest_forecasts=forecasts, quantiles=quantiles, predictions=predictions)
+
+        return predictions
+
+    def _add_forecast_borders(
```

Comment:
What for did we add this method?

Reply:
We extracted this logic from `_forecast_prediction_intervals` to avoid code duplication overwriting this method in `HierarchicalPipeline`